src/adagrams.js

```diff
@@ -1,15 +1,131 @@
 export const drawLetters = () => {
-  // Implement this method for wave 1
+    const LETTER_POOL = {
+      'A': 9, 
+      'B': 2, 
+      'C': 2, 
+      'D': 4, 
+      'E': 12, 
+      'F': 2, 
+      'G': 3, 
+      'H': 2, 
+      'I': 9, 
+      'J': 1, 
+      'K': 1, 
+      'L': 4, 
+      'M': 2, 
+      'N': 6, 
+      'O': 8, 
+      'P': 2, 
+      'Q': 1, 
+      'R': 6, 
+      'S': 4, 
+      'T': 6, 
+      'U': 4, 
+      'V': 2, 
+      'W': 2, 
+      'X': 1, 
+      'Y': 2, 
+      'Z': 1
+      };
+
+    const alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ';
+
+    const hand = [];
+    let letterPool = JSON.parse(JSON.stringify(LETTER_POOL));
+    
+    while(hand.length < 10) {
+        const randomLetter = alphabet[Math.floor(Math.random() * alphabet.length)];
+        if (letterPool[randomLetter] > 0) {
+            letterPool[randomLetter] -= 1;
+            hand.push(randomLetter);
+        }
+    }
+
+    return hand;
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
-};
+        for (let i in input) {
+            if (!(lettersInHand.includes(input[i]))) {
+                return false;
+            } else {
```

Comment:
You can omit the else statement since line 50 and 51 are like a guard clause. If the check doesn't pass on line 50 then we want everything else in the for block to run.

```javascript
for (let i in input) {
    if (!(lettersInHand.includes(input[i]))) {
        return false;
    } 
    const index = lettersInHand.indexOf(input[i]);
    lettersInHand.splice(index, 1);
}